GoMeet/src/test/java/com/google/sps/LocationDaoTest.java

```diff
@@ -0,0 +1,171 @@
+    entity.setProperty("lat", location.getLat());
+    entity.setProperty("lng", location.getLng());
+    entity.setProperty("note", location.getNote());
+    entity.setProperty("voteCount", INIT_VOTE_COUNT);
+    ds.put(entity);
+
+    return KeyFactory.keyToString(entity.getKey());
+  }
+
+  /** Tests if the returns the location entities on the database. */
+  @Test
+  public void getAllTestSingle() {
+    // Set up database
+    String key1 = addLocationToDatabase(LOCATION_A);
+
+    // Expected result
+    Location updatedLocationA =
+        new Location(LOCATION_A.getTitle(), LOCATION_A.getLat(), LOCATION_A.getLng(),
+        LOCATION_A.getNote(), LOCATION_A.getVoteCount(), key1);
+    
+    List<Location> result = ld.getAll(); 
+
+    assertEquals(updatedLocationA, result.get(0));
+  }
+
+  /**
+   *  Tests if getAll() returns all the location entities when there are
+   *  multiples entities in the database.
+   */
+  @Test 
+  public void getAllTestMultiple() {
+    // Set up database
+    String key1 = addLocationToDatabase(LOCATION_A);
+    String key2 = addLocationToDatabase(LOCATION_B);
+
+    List<Location> result = ld.getAll(); 
+    
+    assertEquals(2, result.size());
+  }
+
+  /** Tests if a location is saved to datastore with the correct properties. */
+  @Test
+  public void saveTest() {
+    String keyString = ld.save(LOCATION_A);
+  
+    // Check location entity was added to Datastore
+    assertEquals(1, ds.prepare(new Query("Location")).countEntities(withLimit(10)));
+
+    Query query = new Query("Location");
+    PreparedQuery preparedQuery = ds.prepare(query);
+    Entity result = preparedQuery.asSingleEntity();
+
+    // Check the entity property values were assigned correctly
+    assertEquals(LOCATION_A.getTitle(), result.getProperty("title"));
+    assertEquals(LOCATION_A.getLat(), result.getProperty("lat"));
+    assertEquals(LOCATION_A.getLng(), result.getProperty("lng"));
+    assertEquals(LOCATION_A.getNote(), result.getProperty("note"));
+    assertEquals(LOCATION_A.getVoteCount(), ((Long) result.getProperty("voteCount")).intValue());
+   
+    // Check if the key was returned
+    assertEquals(KeyFactory.keyToString(result.getKey()), keyString);
+  }
+
+  /** Tests that update() adds one to the entity's voteCount. */
+  @Test
+  public void updateTest() {
+    Entity location = new Entity("Location");
+    location.setProperty("title", LOCATION_A.getTitle());
+    location.setProperty("lat", LOCATION_A.getLat());
+    location.setProperty("lng", LOCATION_A.getLng());
+    location.setProperty("note", LOCATION_A.getNote());
+    location.setProperty("voteCount", LOCATION_A.getVoteCount());
+    ds.put(location);
+
+    String keyString = KeyFactory.keyToString(location.getKey());
+
+    try {
+      ld.update(keyString);
+      Entity retrievedLocation = ds.get(location.getKey());
+      assertEquals(2, ((Long) retrievedLocation.getProperty("voteCount")).intValue());
```

Comment:
nit - not sure if its really necessary, but could also check the other properties to ensure they haven't been modified/lost